packages/config/README.md

```diff
@@ -48,52 +48,61 @@ All keys matching `/(config|file|dir)s?$/i` will be treated as a filesystem path
 The following example package.json shows how the config object might look like:
 
 ```JSON
-"name": "my-application",
-"scripts": {
-    "build": "hops build",
-    "develop": "hops develop",
-    "serve": "hops serve",
-    "start": "hops start",
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
-"config": {
-    "hops": {
-      "extends": "path-to-my-hops-config/config.js",
-      "https": true,
-      "host": "0.0.0.0",
-      "port": 3000,
-      "locations": [
-        "/",
-        "/foo"
-      ],
-      "basePath": "my-base-path",
-      "assetPath": "my-asset-path",
-      "browsers": "> 2%, last 1 versions, Firefox ESR",
-      "moduleDirs": [
-        "my-unpublished-packages"
-      ],
-      "appDir": "my-app-directory",
-      "buildDir": "my-build-directory",
-      "cacheDir": "my-cache-directory",
-      "buildConfig": "path-to-my-custom-webpack-config/build.js"
-    }
+{
+  "hops": {
+    "extends": "path-to-my-hops-config/config.js",
+    "https": true,
+    "host": "0.0.0.0",
+    "port": 3000,
+    "locations": [
+      "/",
+      "/foo"
+    ],
+    "basePath": "my-base-path",
+    "assetPath": "my-asset-path",
+    "browsers": "> 2%, last 1 versions, Firefox ESR",
+    "moduleDirs": [
+      "my-unpublished-packages"
+    ],
+    "appDir": "my-app-directory",
+    "buildDir": "my-build-directory",
+    "cacheDir": "my-cache-directory",
+    "buildConfig": "path-to-my-custom-webpack-config/build.js"
   }
+}
 ```
 
-## Configure via CLI
+## Configure via Dedicated File
 
-You can override the configuration values defined in your `package.json`. Consider the above package.json and let's say you want to want to override the `port` value. Note that this only works in an "npm like" context, i.e. with npm and yarn.
+Hops looks for configuration objects in a couple of different places (starting from `process.cwd()`), in this order:
 
-To override the port value for your local environment without actually changing your package.json, set the respective npm environment variable...
+1. `hops` property in your package.json file
+2. `.hopsrc` (or `.hopsrc.(js|json|ya?ml)`)
+3. `hops.config.js` JS file exporting the object
 
-```
-npm config set my-application:hops:port 1337
+You can use whatever suits you best, but please only use a single location in your project's root folder. Hops does not attempt to merge configurations stored in different locations: it uses the first config it finds.
+
+## Environment Specific Configuration
+
+Take the above package.json - and let's say you want to want to override the `port` value. To override the port value for your local environment without changing your application's overall config, add an environment specific config...
+
+```JSON
+{
+  "hops": {
+    "port": 3000,
+    "env": {
+      "my-environment": {
+        "port": 1337
+      }
+    }
+  }
+}
```

Comment:
Should we also add a small section about extending the configuration with a baseline config and how to use placeholders?

Reply:
Also: The readme of the CLI package mentions npm config - that should be rewritten as well.

Reply:
The extension mechanism is already documented - I think what is there should do for now.

Reply:
I'm having trouble finding where the extension mechanism is documented - can you point me to it?